### ISSUE TITLE
[Phase 64.2] SequenceForecaster — Advanced Temporal Forecasting Engine

### ISSUE THREAD

**Issue by web3guru888** [web3guru888/asi-build#1196]
### Overview
The SequenceForecaster implements state-of-the-art forecasting architectures supporting both autoregressive and non-autoregressive prediction, with built-in probabilistic outputs and multi-horizon capability.

#### Transformer-Based Forecasters
- **Temporal Fusion Transformer (TFT)** — Variable selection networks, multi-head attention with interpretable weights, static enrichment, gated residual networks, quantile outputs
- **Informer** — ProbSparse self-attention (O(L log L)), self-attention distilling, generative-style decoder for long-horizon direct prediction
- **Autoformer** — Series decomposition blocks, auto-correlation replacing self-attention, progressive decomposition architecture
- **PatchTST** — Channel-independent patching, instance normalization, supervised and self-supervised pre-training modes
- **FEDformer** — Frequency-enhanced decomposed transformer using Fourier/wavelet bases for seasonal-trend mixing

#### Neural Basis Models
- **N-BEATS** — Doubly residual stacking with interpretable (trend/seasonality) and generic basis functions, ensemble strategies
- **N-HiTS** — Hierarchical interpolation with multi-rate input sampling for different forecast horizons

#### Probabilistic & Autoregressive
- **DeepAR** — Autoregressive RNN with learned likelihood (Gaussian, negative binomial, Student-t), global model across related series
- **Normalizing Flows** — RealNVP and MAF for flexible forecast distribution estimation
- **Conformal Prediction** — Distribution-free prediction intervals with coverage guarantees

#### State-Space Models
- **S4-based forecasting** — Structured state spaces for capturing ultra-long-range dependencies
- **Mamba-TS** — Selective state-space model adapted for temporal forecasting

### Interface Design

```python
class SequenceForecaster:
    def __init__(self, config: ForecasterConfig):
        """Initialize forecaster with model architecture config."""

    async def fit(
        self, train_data: TimeSeriesDataset, val_data: TimeSeriesDataset = None
    ) -> TrainingResult:
        """Train forecasting model on historical data."""

    async def predict(
        self, context: TimeSeries, horizon: int, quantiles: list[float] = [0.1, 0.5, 0.9]
    ) -> ForecastResult:
        """Generate probabilistic forecasts with prediction intervals."""

    async def predict_batch(
        self, contexts: list[TimeSeries], horizon: int
    ) -> list[ForecastResult]:
        """Batch prediction for multiple series."""

    async def backtest(
        self, series: TimeSeries, strategy: BacktestStrategy
    ) -> BacktestResult:
        """Expanding/sliding window backtesting with metrics."""

    async def select_model(
        self, data: TimeSeriesDataset, candidates: list[str]
    ) -> ModelSelectionResult:
        """Automated model selection via cross-validation."""
```

### Evaluation Metrics
- **Point forecasts**: MAE, MSE, RMSE, MAPE, sMAPE, MASE
- **Probabilistic**: CRPS, calibration (coverage), Winkler score, log-likelihood
- **Multi-horizon**: per-horizon breakdown, cumulative metrics

### Integration Points
- **Phase 64.1** (TemporalFeatureExtractor) — Pre-extracted features as model inputs
- **Phase 64.5** (TemporalIntelligenceOrchestrator) — Model lifecycle management
- **Phase 60** (Efficient Inference) — Optimized serving for real-time forecasting
- **Phase 59** (PEFT) — Efficient fine-tuning of temporal foundation models

### Testing Strategy
- Unit tests for each architecture (TFT, Informer, N-BEATS, DeepAR)
- Integration tests: train → predict → evaluate pipeline
- Benchmark on standard datasets (ETTh1/h2, Weather, Electricity, Traffic)
- Probabilistic calibration tests (coverage vs nominal level)
- Edge cases: cold-start, missing values, exogenous variable handling

### References
- Lim et al. (2021). "Temporal Fusion Transformers for Interpretable Multi-horizon Time Series Forecasting"
- Zhou et al. (2021). "Informer: Beyond Efficient Transformer for Long Sequence Time-Series Forecasting"
- Wu et al. (2021). "Autoformer: Decomposition Transformers with Auto-Correlation"
- Nie et al. (2023). "A Time Series is Worth 64 Words"
- Oreshkin et al. (2020). "N-BEATS: Neural Basis Expansion Analysis"
- Salinas et al. (2020). "DeepAR: Probabilistic Forecasting with Autoregressive Recurrent Networks"
